src/client.rs

```diff
@@ -255,14 +255,22 @@ impl ServerCertVerifier for Verifier {
         &self,
         end_entity: &CertificateDer,
         intermediates: &[CertificateDer],
-        server_name: &rustls::ServerName,
+        server_name: &pki_types::ServerName<'_>,
         ocsp_response: &[u8],
         _now: UnixTime,
     ) -> Result<ServerCertVerified, rustls::Error> {
         let cb = self.callback;
         let server_name: Cow<'_, str> = match server_name {
-            rustls::ServerName::DnsName(n) => n.as_ref().into(),
-            rustls::ServerName::IpAddress(ip) => ip.to_string().into(),
+            pki_types::ServerName::DnsName(n) => n.as_ref().into(),
+            // TODO(@cpu): HACK
```

Comment:
Trying to think through the right change to make this more ergonomic. I think `pki_types::IpAddr` should `impl From<IpAddr> for std::net::IpAddr ? :thinking:

Reply:
https://github.com/rustls/pki-types/pull/14